component/plugincontroller/component.go

```diff
@@ -0,0 +1,34 @@
+package plugincontroller
+
+import (
+	"github.com/docker/docker/component"
+	"github.com/docker/docker/plugin"
+	"golang.org/x/net/context"
+)
+
+const name = "plugincontroller"
+
+// Set sets the plugin controller component
+func Set(s *plugin.Manager) (cancel func(), err error) {
+	return component.Register(name, s)
+}
```

Comment:
"Very small coordinator", except it needs to import packages for every single subsystem (and often some of the dependencies of those subsystems).

I don't know about you, but I can barely work in the `daemon` package just simply because it imports too much and brings my editor to a screeching halt, this is nothing to do with the amount of logic it holds but rather the number of things it's initializing, interacting with, etc.